stl/inc/tuple

```diff
@@ -724,6 +745,42 @@ _NODISCARD constexpr bool operator==(const tuple<_Types1...>& _Left, const tuple
     return _Left._Equals(_Right);
 }
 
+#ifdef __cpp_lib_concepts
+template <class... _Types1, class... _Types2>
+_NODISCARD constexpr auto operator<=>(const tuple<_Types1...>& _Left, const tuple<_Types2...>& _Right) {

```

Comment:
Using return type deduction here is not equivalent to the depicted `common_comparison_category_t<synth-three-way-result<TTypes, UTypes>...>` in the Standard; the latter will error in the immediate context instantiation when substituting in `_Types1` and `_Types2`, but the former will not. That means things like e.g. `three_way_comparable_with<tuple<apple>, tuple<orange>>` could hard error instead of evaluate to `false`.

This suggests we're missing test coverage for such negative cases.